source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -286,63 +288,95 @@ Supported Evergreen variants that have the AWS SAM CLI installed:
 - rhel82-arm64
 
 
-This is an example function in the Evergreen config that accomplishes this, using
-subprocess.exec to execute a script that calls the drivers-evergreen-tools
-function inside of it:
+This is an example task group in the Evergreen config that accomplishes this, using
+subprocess.exec to execute scripts that call the drivers-evergreen-tools
+functions inside of it for setup, teardown, and execution:
 
 .. code:: yaml
 
-  run deployed aws lambda tests:
-  - command: ec2.assume_role
-    params:
-      role_arn: ${LAMBDA_AWS_ROLE_ARN}
-  - command: subprocess.exec
-    params:
-      working_dir: src
-      binary: bash
-      args:
-        - .evergreen/run-deployed-lambda-aws-tests.sh
-      env:
-        TEST_LAMBDA_DIRECTORY: ${PROJECT_DIRECTORY}/test/lambda
-        DRIVERS_TOOLS: ${DRIVERS_TOOLS}
-        DRIVERS_ATLAS_PUBLIC_API_KEY: ${DRIVERS_ATLAS_PUBLIC_API_KEY}
-        DRIVERS_ATLAS_PRIVATE_API_KEY: ${DRIVERS_ATLAS_PRIVATE_API_KEY}
-        DRIVERS_ATLAS_LAMBDA_USER: ${DRIVERS_ATLAS_LAMBDA_USER}
-        DRIVERS_ATLAS_LAMBDA_PASSWORD: ${DRIVERS_ATLAS_LAMBDA_PASSWORD}
-        DRIVERS_ATLAS_GROUP_ID: ${DRIVERS_ATLAS_GROUP_ID}
-        LAMBDA_STACK_NAME: dbx-node-lambda
-        AWS_REGION: us-east-1
-        AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
-        AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
-        AWS_SESSION_TOKEN: ${AWS_SESSION_TOKEN}
+  tasks:
+    - name: test-aws-lambda-deployed
+      commands:
+        - func: install dependencies
+        - command: ec2.assume_role
+          params:
+            role_arn: ${LAMBDA_AWS_ROLE_ARN}
+            duration_seconds: 3600
+        - command: subprocess.exec
+          params:
+            working_dir: src
+            binary: bash
+            args:
+              - ${DRIVERS_TOOLS}/.evergreen/aws_lambda/run-deployed-lambda-aws-tests.sh
+            env:
+              MONGODB_URI: ${MONGODB_URI}
```

Comment:
Can we use an expansion to avoid repeating all of these values?

Reply:
Updated to add expansions to the env.